### ISSUE TITLE
Char.IsXXX predicates give compile error

### ISSUE THREAD

**Issue by jcoder58** [fable-compiler/Fable#858]
### Description

System.Char.IsWhiteSpace (and other IsXXX) generate compile errors.

### Repro code

Start with the default Fable template

    module test
    let x = System.Char.IsWhiteSpace

### Expected and actual results

    ERROR in ./src/App.fs
    src\App.fs(2,8,2,32) : error FABLE: Cannot find replacement for System.Char::IsWhiteSpace

### Related information 

* fable-compiler version: 1.0.0 beta
* fable-core version: 1.0.0 beta
* Operating system: Windows 8.1

**Comment by alfonsogarciacaro** [comment]
Ups, I didn't know about those methods. Would you like to send a PR to support them? I can guide you about how to add new replacements.

**Comment by jcoder58** [comment]
I looked into this and there is a problem with implementing these.  The Char.IsXXX are based on Unicode classes, whereas Javascript doesn't support testing for Unicode class types.   [This StackOverflow answer](http://stackoverflow.com/a/25850689/167922) shows what's involved in implementing Char.IsLetter.  

There is a need for this for validation, parsing and other applications, but do you want this in Fable.Core?

**Comment by Pauan** [comment]
The [xregexp](https://github.com/slevithan/xregexp) library seems to have a comprehensive [set of regexps for testing Unicode categories](https://github.com/slevithan/xregexp/blob/master/src/addons/unicode-categories.js)

It's huge and complicated, but maybe we can rely on tree shaking to remove it when it's unused?

**Comment by jcoder58** [comment]
I found [this page](http://rxaviers.github.io/javascript-globalization/) which lists libraries that handle more globalization issues then just the unicode char sets.    

Also, [this respository](https://github.com/mathiasbynens/unicode-9.0.0#readme) also has data for all the character classes in a number of formats, including regex.   He has one for each version of Unicode, so if you want to match what .NET uses, it is possible.

**Comment by ncave** [comment]
@alfonsogarciacaro There is basic support for unicode in [FCS-fable](https://github.com/ncave/FSharp.Compiler.Service/blob/fable/src/fsharp/fcs-fable/adapters.fs#L97-L137) ([latin1](https://github.com/ncave/FSharp.Compiler.Service/blob/fable/src/fsharp/fcs-fable/unicode.fs) categories, no surrogate pairs) that can be extended and pushed downstream to Fable-Core if needed.

**Comment by alfonsogarciacaro** [comment]
@ncave That's a very good day. I'll add F#>JS compilation in the fable-core build process so it's possible to PR it :+1:

@jcoder58 Supporting the whole BCL semantics is not a goal in Fable. Sometimes it's not possible in JS (concurrency) or it's too big an undertaking (globalization), but we do want to extend BCL support and probably there's an opportunity here to improve unicode support (at least for latin sets). But I don't remember well how .NET deals with unicode, could you please PR some unit tests so it's easier to see what you need exactly?

**Comment by alfonsogarciacaro** [comment]
@jcoder58 would you be interested in helping to implement this?

**Comment by ncave** [comment]
@alfonsogarciacaro What is needed for the F#>JS compilation in the fable-core build process to happen? Does it depend on #800 to be done first?

**Comment by alfonsogarciacaro** [comment]
@ncave Yes, implementing the features listed [in this comment](https://github.com/fable-compiler/Fable/issues/800#issuecomment-299986774) is necessary so we can add F# files to fable-core without bundling

**Comment by alfonsogarciacaro** [comment]
This is already supported :+1:
